test/integration/webhook_test.go

```diff
@@ -759,3 +604,110 @@ func TestValidationWebhook(t *testing.T) {
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "some fields were invalid due to missing data: rsa_public_key, key, secret")
 }
+
+func ensureWebhookService(name string) (func() error, error) {
+	validationsService, err := env.Cluster().Client().CoreV1().Services(controllerNamespace).Create(ctx, &corev1.Service{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "Service"},
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "default",
+					Port:       443,
+					TargetPort: intstr.FromInt(49023),
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("creating webhook service: %w", err)
+	}
+
+	nodeName := "aaaa"
+	endpoints, err := env.Cluster().Client().CoreV1().Endpoints(controllerNamespace).Create(ctx, &corev1.Endpoints{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "Endpoints"},
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Subsets: []corev1.EndpointSubset{
+			{
+				Addresses: []corev1.EndpointAddress{
+					{
+						IP:       "172.17.0.1",
+						NodeName: &nodeName,
+					},
+				},
+				Ports: []corev1.EndpointPort{
+					{
+						Name:     "default",
+						Port:     49023,
+						Protocol: corev1.ProtocolTCP,
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("creating webhook endpoints: %w", err)
+	}
+
+	closer := func() error {
+		if err := env.Cluster().Client().CoreV1().Services(controllerNamespace).Delete(ctx, validationsService.Name, metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+
+		if err := env.Cluster().Client().CoreV1().Endpoints(controllerNamespace).Delete(ctx, endpoints.Name, metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	return closer, nil
+}
+
+func waitForWebhookService(t *testing.T) {
+	require.Eventually(t, func() bool {
+		_, err := net.DialTimeout("tcp", "172.17.0.1:49023", 1*time.Second)
```

Comment:
My hope would be that we wont be using this `kind` specific hack much longer, but while we're here just in case it is for a while longer I think now is a good time to create a `const` out of this address with a bit of `godoc` above it that explains what it is and provides some insights about its `kind`-only nature.

Reply:
0df648a7ca2ffcb32a82c24837b0b75346cf2ae7

A plain old comment instead of a `godoc` because it's a broader explanation.